### PR TITLE
define diag ports in helm

### PR DESCRIPTION
**Pull request by dinesh**
Our monitoring agent (signalfx-agent) discovers pods when we actually define them in the object spec. This PR adds the diag port (3000) for high-availability setup.

To test -
 
```
$ helm template examples/chart/teleport --set config.highAvailability=true | grep -C 2 diag
        - name: diag
          containerPort: 3000
--
        - name: authssh
          containerPort: 3025
        - name: diag
          containerPort: 3000
        resources:
```

### CODE REVIEW

**Comment by webvictim** [comment]
@dinesh This port should be optional i.e. gated behind a feature flag, as the chart doesn't open port 3000 by default.

**Review by webvictim — CHANGES_REQUESTED** [review]
@dinesh This port should be optional i.e. gated behind a feature flag, as the chart doesn't open port 3000 by default.

**Comment by dinesh** [comment]
@webvictim tried adding logic to detect if metrics are enabled though it's a bit hacky. Please take a look.

File: examples/chart/teleport/templates/deployment.yaml [hunk 1]
```diff
@@ -178,6 +178,12 @@ spec:
         ports:
         - name: authssh
           containerPort: {{ .Values.ports.authssh.containerPort }}
+        {{- range .Values.extraArgs }}
+        {{- if contains "--diag-addr" .}}
```
Comment (webvictim) [inline examples/chart/teleport/templates/deployment.yaml:182]:
```suggestion
        {{- if contains "--diag-addr" . }}
```

**Review by webvictim — COMMENTED** [review]
That's actually not a bad way of doing it. Maybe it should check for `--diag-addr=0.0.0.0:3000` instead? If you specify `--diag-addr=127.0.0.1:3000` then the port doesn't need to be exposed because it's only listening for loopback connections. Seems like we could be explicit about only exposing it when the container's actually going to be listening on all interfaces.

File: examples/chart/teleport/templates/deployment.yaml (hunk 1 above)
Comment (dinesh) [reply examples/chart/teleport/templates/deployment.yaml]: cool makes sense. updated it.

**Review by webvictim — APPROVED** [review]

**Comment by webvictim** [comment]
Our CI doesn't run tests on external PRs at the moment so I'm going to make a branch and PR it myself.

**Comment by webvictim** [comment]
My PR: https://github.com/gravitational/teleport/pull/7212

**Comment by webvictim** [comment]
Hey @dinesh, thanks for the PR. It's been merged to `master` now, see https://github.com/gravitational/teleport/pull/7212

Closing this mirror.

**Pull request closed without merging**